Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -496,6 +503,14 @@ public struct DBPointer: BSONValue, Codable, Equatable {
     }
 }
 
+// An extension of `DBPointer` to add capability to be hashed
+extension DBPointer: Hashable {
+    public func hash(into hasher: inout Hasher) {
```

Comment:
I think the comment Patrick made last time about the synthesized implementation applies here - since this is just hashing all the properties, we don't need to write the function ourselves and can just mark the type as `Hashable`.
this should also be true for a few more types below